gnc/sub8_thruster_mapper/nodes/mapper.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+import rospy
+from scipy.optimize import minimize
+import numpy as np
+from sub8_ros_tools import wait_for_param, thread_lock, rosmsg_to_numpy
+import threading
+from sub8_msgs.msg import Thrust, ThrusterCmd
+from geometry_msgs.msg import WrenchStamped
+
+
+lock = threading.Lock()
+class ThrusterMapper(object):
+    def __init__(self, layout):
+        '''The layout should be a dictionary of the form used in thruster_mapper.launch
+        an excerpt is quoted here for posterity
+
+        busses:
+            - port: /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A403IMC9-if00-port0
+              thrusters:
+                FLV: {
+                  node_id: 10,
+                  frame_id: /base_link,
+                  position: [0.1583, 0.169, 0.0142],
+                  direction: [0, 0, -1]
+                }
+                FLL: {
+                  node_id: 11,
+                  frame_id: /base_link,
+                  position: [0.2678, 0.2795, 0],
+                  direction: [-0.866, 0.5, 0]
+                }
+
+            - port: ....
+              thrusters: ....
+
+        '''
+        self.min_thrust, self.max_thrust = self.get_ranges()
+        # Make thruster layout
+        self.B = self.generate_B(layout)
+        self.wrench_sub = rospy.Subscriber('/wrench', WrenchStamped, self.request_wrench_cb, queue_size=1)
+        self.thruster_pub = rospy.Publisher('/thrust', Thrust, queue_size=1)
+
+    @thread_lock(lock)
+    def update_layout(self, srv):
+        '''Respond to a thruster-out alarm'''
```

Comment:
The name of this function and the docstring are a bit confusing. Is the sole purpose of this function for responding to a thruster-out alarm?